pydantic/functional_validators.py

```diff
@@ -163,25 +163,45 @@ class PlainValidator:
 
     A metadata class that indicates that a validation should be applied **instead** of the inner validation logic.
 
+    !!! note:
+        Before v2.9, `PlainValidator` wasn't always compatible with JSON Schema generation for `mode='validation'`.
+        You can now use the `json_schema_input_type` argument to specify the input type of the function
+        to be used in the JSON schema when `mode='validation'` (the default). See the example below for more details.
+
+        If you'd like the JSON Schema type for an annotated field to be populated automatically,
+        you might consider using `WrapValidator` instead.
```

Comment:
Maybe we shouldn't recommend a specific validator type. before and wrap validators both populate the JSON Schema from the annotated type, but the usage of one of the user will depend on the user